### PR TITLE
Python: Allow function calling stepwise planner to use AzureOpenAI chat service

### PR DESCRIPTION
**Pull request by moonbox3**
### Motivation and Context

The first iteration of the function calling stepwise planner only used OpenAI. Now allow AzureChatService to use the planner.

### Description

This PR:
- Allows the use of AzureOpenAI for the FC Stepwise Planner
- Fixes some formatting issues when generating the names/metadata of Plugins. .NET gets the function metadata as JSON, and the Azure Model performs much better with it formatted like this.
- Fix some missing spaces in the constant message used in the planner -- Azure is treating the new lines in that const message incorrectly and it causes content filtering results (reaching out to the team to understand why this is).

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/8e0bd9c5e35850e34de2232e5d4e0301513145df/README.md">Py3.8 Test Coverage</a><details><summary>Python 3.8 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td colspan="5"><b>semantic_kernel/core_plugins</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/8e0bd9c5e35850e34de2232e5d4e0301513145df/python/semantic_kernel/core_plugins/time_plugin.py">time_plugin.py</a></td><td>84</td><td>6</td><td>93%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/8e0bd9c5e35850e34de2232e5d4e0301513145df/python/semantic_kernel/core_plugins/time_plugin.py#L61">61</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/8e0bd9c5e35850e34de2232e5d4e0301513145df/python/semantic_kernel/core_plugins/time_plugin.py#L71-L72">71&ndash;72</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/8e0bd9c5e35850e34de2232e5d4e0301513145df/python/semantic_kernel/core_plugins/time_plugin.py#L181-L182">181&ndash;182</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/8e0bd9c5e35850e34de2232e5d4e0301513145df/python/semantic_kernel/core_plugins/time_plugin.py#L235">235</a></td></tr><tr><td colspan="5"><b>semantic_kernel/planners</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/8e0bd9c5e35850e34de2232e5d4e0301513145df/python/semantic_kernel/planners/planner_extensions.py">planner_extensions.py</a></td><td>32</td><td>14</td><td>56%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/8e0bd9c5e35850e34de2232e5d4e0301513145df/python/semantic_kernel/planners/planner_extensions.py#L20-L26">20&ndash;26</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/8e0bd9c5e35850e34de2232e5d4e0301513145df/python/semantic_kernel/planners/planner_extensions.py#L30-L31">30&ndash;31</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/8e0bd9c5e35850e34de2232e5d4e0301513145df/python/semantic_kernel/planners/planner_extensions.py#L44-L51">44&ndash;51</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/8e0bd9c5e35850e34de2232e5d4e0301513145df/python/semantic_kernel/planners/planner_extensions.py#L59-L70">59&ndash;70</a></td></tr><tr><td colspan="5"><b>semantic_kernel/planners/function_calling_stepwise_planner</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/8e0bd9c5e35850e34de2232e5d4e0301513145df/python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner.py">function_calling_stepwise_planner.py</a></td><td>93</td><td>34</td><td>63%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/8e0bd9c5e35850e34de2232e5d4e0301513145df/python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner.py#L115-L169">115&ndash;169</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/8e0bd9c5e35850e34de2232e5d4e0301513145df/python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner.py#L200-L205">200&ndash;205</a></td></tr><tr><td><b>TOTAL</b></td><td><b>5369</b></td><td><b>1014</b></td><td><b>81%</b></td><td>&nbsp;</td></tr></tbody></table></details>

## Python 3.8 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1190 | 11 :zzz: | 0 :x: | 0 :fire: | 19.860s :stopwatch: |

File: python/semantic_kernel/planners/planner_extensions.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 import logging
 
+from semantic_kernel.connectors.ai.open_ai.utils import get_function_calling_object
```
Comment (juliomenendez) [inline python/semantic_kernel/planners/planner_extensions.py:5]: Probably don't need this neither

**Review by juliomenendez — CHANGES_REQUESTED** [review]
Looks good, just one small extra import that I think is not needed

File: python/semantic_kernel/planners/planner_extensions.py (hunk 1 above)
Comment (moonbox3) [reply python/semantic_kernel/planners/planner_extensions.py]: thanks done

**Review by juliomenendez — APPROVED** [review]

**Review by eavanvalkenburg — APPROVED** [review]

**Pull request merged**